core/client/fs/src/main/java/alluxio/client/block/stream/NettyPacketWriter.java

```diff
@@ -240,6 +249,23 @@ public void flush() throws IOException {
       Thread.currentThread().interrupt();
       throw new CanceledException(e);
     }
+    if (mEOFSent || mCancelSent || pos == 0) {
```

Comment:
The reasons I add `pos==0` here is:
(1) Flush is meaningless if no write happened.
(2) we are not able to reply flush directly when we detect a FLUSH in worker side.
netty I/0 thread responsible for reading the messages from client.
packet writer thread responsible for executing and replying the messages.
(3) because of (2), FLUSH will have to choose to `createRequestContext` and `initRequestContext` or not in the netty I/O thread. Both are very error-prone, I chose to let the FLUSH create and init context, then write will give an exception: TempBlockMeta not found for BlockID....

To reduce the meaningless operations(reason 1) and difficulty (reason 2 & 3), this check is an optimization in my thought.